dimelo/parse_bam.py

```diff
@@ -227,17 +238,18 @@ def pileup(
             print(
                 "No base modification threshold provided. Using adaptive threshold selection via modkit."
             )
-    elif thresh <= 0:
-        raise ValueError(
-            f"Threshold {thresh} cannot be used for pileup, please pick a positive nonzero value."
-        )
     else:
         adjusted_threshold = utils.adjust_threshold(thresh, quiet=quiet)
-        for modnames_set in utils.BASEMOD_NAMES_DICT.values():
-            for modname in modnames_set:
+        if adjusted_threshold < 0.5 and not quiet:
+            print(
+                f"WARNING: thresh {thresh} is very low and may lead to unexpected behavior. Typical thresholds are at least 0.5 or 128."
+            )
+        for motif in motifs:
+            parsed_motif = utils.ParsedMotif(motif)
```

Comment:
Wait why are we parsing motifs again in the same method? Can we get away with parsing once and referencing saved data?

Reply:
Yes we should probably adjust this. I have it this way to keep different parts of parse_bam functions as modular as possible, given that parsing a motif takes next to no time anyway. However, agreed that we should probably just parse them all up top and then reference the parsed list.